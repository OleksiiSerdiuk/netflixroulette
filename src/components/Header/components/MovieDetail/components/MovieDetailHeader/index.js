export { default as MovieDetailHeader } from './MovieDetailHeader';
