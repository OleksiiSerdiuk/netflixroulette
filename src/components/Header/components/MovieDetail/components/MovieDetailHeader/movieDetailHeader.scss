.movie-detail-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 0;
  position: relative;
  z-index: 2;
}

.btn--search-icon {
  min-height: 48px;
}