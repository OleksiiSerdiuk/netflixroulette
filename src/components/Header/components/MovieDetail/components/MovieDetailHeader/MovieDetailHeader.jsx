import SearchIcon from '../../../../../../images/searchIcon.svg';
import Button from '../../../../../common/Button';
import './movieDetailHeader.scss';
import { MovieContext } from '../../../../../../contexts';
import { useContext } from 'react';
import { Logo } from '../../../../../Logo/Logo';
import { useNavigate } from 'react-router-dom';

const MovieDetailHeader = () => {
  const { setMovieDescription } = useContext(MovieContext);
  const navigate = useNavigate();

  const handlerReturnToFindMovie = () => {
    setMovieDescription(null);
    navigate(`/search`);
  };

  return (
    <div className='movie-detail-header'>
      <Logo />
      <Button
        btnClass='btn-default btn--search-icon'
        btnText={<img src={SearchIcon} alt='search' />}
        click={handlerReturnToFindMovie}
        type={'submit'}
      />
    </div>
  );
};

export default MovieDetailHeader;
