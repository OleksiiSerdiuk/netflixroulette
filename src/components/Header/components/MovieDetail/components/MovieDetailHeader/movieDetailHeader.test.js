import { fireEvent, render, waitFor } from '@testing-library/react';
import MovieDetailHeader from './MovieDetailHeader';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

const handleSubmit = jest.fn();

describe('movie detail header', () => {
  it('renders movie detail header snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <MovieDetailHeader />
      </BrowserRouter>
    );
    expect(asFragment(<MovieDetailHeader />)).toMatchSnapshot();
  });

  it('should handle submit', async () => {
    const { getByRole } = render(
      <BrowserRouter>
        <MovieDetailHeader />
      </BrowserRouter>
    );

    await act(() => {
      fireEvent.click(getByRole('button', { type: /submit/i }));
      handleSubmit();
    });
    await waitFor(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(1);
    });
  });
});
