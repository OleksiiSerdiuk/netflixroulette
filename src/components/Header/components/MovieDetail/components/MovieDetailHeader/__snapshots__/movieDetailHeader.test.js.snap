// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie detail header renders movie detail header snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-detail-header"
  >
    <img
      alt="Logo"
      src="test-file-stub"
    />
    <button
      class="btn btn-default btn--search-icon"
      type="submit"
    >
      <img
        alt="search"
        src="test-file-stub"
      />
    </button>
  </div>
</DocumentFragment>
`;
